src/framework/MOM_diag_mediator.F90

```diff
@@ -1667,6 +1677,17 @@ subroutine diag_mediator_init(G, param_file, diag_cs, err_msg, doc_file_dir)
   diag_cs%isd = G%isd ; diag_cs%ied = G%ied
   diag_cs%jsd = G%jsd ; diag_cs%jed = G%jed
 
+  call get_param(param_file,mod,"APPLY_DIAGNOSTIC_MASK_FOR_ZSTAR",diag_cs%apply_3d_diag_mask_for_zstar_only, &
```

Comment:
How about we rename parameter to something like "ASSUME_DIAG_3DMASK_IS_STATIC" which would allow for other coordinates (e.g. terrain-following, eta-coordinate) where the numbers of degrees of freedom are constant-in-time for each column.